### ISSUE TITLE
[FEATURE] Adding current location button

### ISSUE THREAD

**Issue by harsh5902** [valhalla/web-app#131]
### What kind of feature you are suggesting?
I would like to add a button which will move the map to user's location. 
### Does it solve any problem?
Usually user tends to locate himself/herself or try to know the distance from his location to other.

**Comment by nilsnolde** [comment]
I just wanted to close and reference another issue asking for the same thing in #85. But then I realized you were actually the one already asking again for the same thing in https://github.com/gis-ops/valhalla-app/issues/105 :laughing:

We now remember the last map extent (seems like that was also you in https://github.com/gis-ops/valhalla-app/issues/120), I think that's quite convenient too. I could be convinced to ask for a user's location, but for that enough users will have to upvote this issue. So let me leave this open for now.

**Comment by harsh5902** [comment]
Yes, I do understand you merged my pull request recently #120 (last map extent). While working with that issue I realized that you wanted to maintain user's confidentiality and security. But sometimes user wants to locate himself at unknown location or usual menality of the user is to know distace or manuvers to get to other location from his own.

So, for this we can add a prompt for user wether he is comfortable sharing his realtime routing status (or GPS we say) with the app. That would be better solution for it I guess.

**Comment by harsh5902** [comment]
Though I feel that even the last extent of the map is also good but this feature could get so much necessary sometimes

Do let me know your thoughts...

**Comment by nilsnolde** [comment]
Yeah I guess you're right, it doesn't need to ask for permission when loading the app, but rather on user request via some 
 button (ideally in the button toolbar on the right), at which point it'd be the user's decision. So yeah, in the end, I'm with you I think.

**Comment by harsh5902** [comment]
So, suggest what to do, shall I wait for other users feedback or get to start working on it??

**Comment by nilsnolde** [comment]
You can if you want. I'm not restricting anyone to work on anything they found themselves, but IMO it's not really necessary to deal with that now, it could be done within the GSoC project (or after).

**Comment by harsh5902** [comment]
I am on it @ganesh1828 just little busy with clg work.

**Comment by spyatmycode** [comment]
> I am on it @ganesh1828 just little busy with clg work.

How is it going so far? Have you resolved this issue?

**Comment by harsh5902** [comment]
@nilsnolde I will raise the PR in 5-6 days time, as currently I have university exams going on. That is why it is taking time.
